File: 766_Toeplitz_Matrix.js
var isToeplitzMatrix = function(matrix) {
    if(matrix[0].length == 1) return false
    let firstElement = matrix[0][0]
    for(let i = 0; i < matrix.length; i++){
        if(firstElement != matrix[i][i]) return false
    }
    return true
};

File: 766_Toeplitz_Matrix.cpp
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        int i = 1;
        while(i < matrix.size()){
            int j = 1;
            while(j < matrix[0].size()){
                if(matrix[i][j] != matrix[i - 1][j - 1]){
                    return false;
                }
                j++;
            }
            i++;
        }
        return true;
    }
};
